config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/TrustedCertEntityBuilder.java

```diff
@@ -72,7 +72,9 @@
             case DEPLOYMENT:
                 return entities.entrySet().stream()
                         .map(
-                                trustedCertEntry -> EntityBuilderHelper.getEntityWithOnlyMapping(TRUSTED_CERT_TYPE, trustedCertEntry.getKey(), idGenerator.generate())
+                                trustedCertEntry -> EntityBuilderHelper.getEntityWithOnlyMapping(TRUSTED_CERT_TYPE, trustedCertEntry.getKey(),
+                                        ((TrustedCert)trustedCertEntry.getValue()).getAnnotatedEntity() != null && ((TrustedCert)trustedCertEntry.getValue()).getAnnotatedEntity().getId() != null ?
```

Comment:
Please move this to a variable. It gets unreadable and difficult to understand the expression.

Reply:
done